discovery/uyuni/uyuni.go

```diff
@@ -0,0 +1,343 @@
+	}
+	return nil
+}
+
+// Attempt to login in Uyuni Server and get an auth token
+func login(rpcclient *xmlrpc.Client, user string, pass string) (string, error) {
+	var result string
+	err := rpcclient.Call("auth.login", []interface{}{user, pass}, &result)
+	return result, err
+}
+
+// Logout from Uyuni API
+func logout(rpcclient *xmlrpc.Client, token string) error {
+	err := rpcclient.Call("auth.logout", token, nil)
+	return err
+}
+
+// Get the system groups information of monitored clients
+func getSystemGroupsInfoOfMonitoredClients(rpcclient *xmlrpc.Client, token string) (map[int][]systemGroupID, error) {
+	var systemGroupsInfos []struct {
+		SystemID     int             `xmlrpc:"id"`
+		SystemGroups []systemGroupID `xmlrpc:"system_groups"`
+	}
+	err := rpcclient.Call("system.listSystemGroupsForSystemsWithEntitlement", []interface{}{token, monitoringEntitlementLabel}, &systemGroupsInfos)
+	if err != nil {
+		return nil, err
+	}
+	result := make(map[int][]systemGroupID)
+	for _, systemGroupsInfo := range systemGroupsInfos {
+		result[systemGroupsInfo.SystemID] = systemGroupsInfo.SystemGroups
+	}
+	return result, nil
+}
+
+// GetSystemNetworkInfo lists client FQDNs
+func getNetworkInformationForSystems(rpcclient *xmlrpc.Client, token string, systemIDs []int) (map[int]networkInfo, error) {
+	var networkInfos []networkInfo
+	err := rpcclient.Call("system.getNetworkForSystems", []interface{}{token, systemIDs}, &networkInfos)
+	if err != nil {
+		return nil, err
+	}
+	result := make(map[int]networkInfo)
+	for _, networkInfo := range networkInfos {
+		result[networkInfo.SystemID] = networkInfo
+	}
+	return result, nil
+}
+
+// Get endpoints information for given systems
+func getEndpointInfoForSystems(
+	rpcclient *xmlrpc.Client,
+	token string,
+	systemIDs []int,
+) ([]endpointInfo, error) {
+	var endpointInfos []endpointInfo
+	err := rpcclient.Call(
+		"system.monitoring.listEndpoints",
+		[]interface{}{token, systemIDs}, &endpointInfos)
+	if err != nil {
+		return nil, err
+	}
+	return endpointInfos, err
+}
+
+// NewDiscovery returns a new file discovery for the given paths.
+func NewDiscovery(conf *SDConfig, logger log.Logger) *Discovery {
+	d := &Discovery{
+		interval: time.Duration(conf.RefreshInterval),
+		sdConfig: conf,
+		logger:   logger,
+	}
+	d.Discovery = refresh.NewDiscovery(
+		logger,
+		"uyuni",
+		time.Duration(conf.RefreshInterval),
+		d.refresh,
+	)
+	return d
+}
+
+func (d *Discovery) getEndpointLabels(
+	endpoint endpointInfo,
+	systemGroupIDs []systemGroupID,
+	networkInfo networkInfo,
+) model.LabelSet {
+
+	var hostname string
+	var addr string
+	managedGroupNames := getSystemGroupNames(systemGroupIDs)
+	if len(networkInfo.PrimaryFQDN) > 0 {
+		hostname = networkInfo.PrimaryFQDN
+	} else {
+		hostname = networkInfo.Hostname
+	}
+	if endpoint.Port > 0 {
+		addr = fmt.Sprintf("%s:%d", hostname, endpoint.Port)
+	} else {
+		addr = hostname
```

Comment:
We should add a Port parameter to the SD to ack as default, like we do on other SD's

Reply:
Port number has to be in fact always available in `endpointInfo`. Changed accordingly.